aten/src/ATen/CPUApplyUtils.h

```diff
@@ -179,6 +180,17 @@ inline bool iterate_continue(Arg& iter, Args&... iter_tail) {
       iterate_continue(iter_tail...);
 }
 
+inline int64_t max_iterate_size() {
+  return std::numeric_limits<int64_t>::max();
+};
+
+template <typename Arg, typename... Args>
+inline int64_t max_iterate_size(Arg& iter, Args&... iter_tail) {
```

Comment:
I think in one of our conversations we agreed that this could be determined statically before the loop even starts to run, and we could avoid recomputing this. Why do we do it like that now?

Reply:
Yes, I'll revisit this as well towards the end.

Reply:
Do you still want to revisit that?

Reply:
Done